Source/Renderer/ComputeEngine.js

```diff
@@ -0,0 +1,213 @@
+    var drawCommandScratch = new DrawCommand({
+        primitiveType : PrimitiveType.TRIANGLES
+    });
+    var clearCommandScratch = new ClearCommand({
+        color : new Color(0.0, 0.0, 0.0, 0.0)
+    });
+
+    function getViewportQuadVertexArray(computeEngine) {
+        var vertexArray = computeEngine._viewportQuadVertexArray;
+
+        if (!defined(vertexArray)) {
+            var geometry = new Geometry({
+                attributes : {
+                    position : new GeometryAttribute({
+                        componentDatatype : ComponentDatatype.FLOAT,
+                        componentsPerAttribute : 2,
+                        values : [
+                            -1.0, -1.0,
+                            1.0, -1.0,
+                            1.0,  1.0,
+                            -1.0,  1.0
+                        ]
+                    }),
+
+                    textureCoordinates : new GeometryAttribute({
+                        componentDatatype : ComponentDatatype.FLOAT,
+                        componentsPerAttribute : 2,
+                        values : [
+                            0.0, 0.0,
+                            1.0, 0.0,
+                            1.0, 1.0,
+                            0.0, 1.0
+                        ]
+                    })
+                },
+                // Workaround Internet Explorer 11.0.8 lack of TRIANGLE_FAN
+                indices : new Uint16Array([0, 1, 2, 0, 2, 3]),
+                primitiveType : PrimitiveType.TRIANGLES
+            });
+
+            vertexArray = VertexArray.fromGeometry({
+                context : computeEngine._context,
+                geometry : geometry,
+                attributeLocations : {
+                    position : 0,
+                    textureCoordinates : 1
+                },
+                bufferUsage : BufferUsage.STATIC_DRAW,
+                interleave : true
+            });
+
+            computeEngine._viewportQuadVertexArray = vertexArray;
+        }
+
+        return vertexArray;
+    }
+
+    function createFramebuffer(context, texture) {
+        var fbo = new Framebuffer({
+            context : context,
+            colorTextures : [texture]
+        });
+        fbo.destroyAttachments = false;
+        return fbo;
+    }
+
+    function createViewportQuadShader(context, fragmentShaderSource) {
+        return ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : ViewportQuadVS,
+            fragmentShaderSource : fragmentShaderSource,
+            attributeLocations : viewportQuadAttributeLocations
+        });
+    }
+
+    function createRenderState(width, height) {
+        if ((!defined(renderStateScratch)) ||
+            (renderStateScratch.viewport.width !== width) ||
+            (renderStateScratch.viewport.height !== height)) {
+
+            renderStateScratch = RenderState.fromCache({
+                viewport : new BoundingRectangle(0, 0, width, height)
```

Comment:
Can we just use a default render state since the pass state with override `width` and `height` when it is undefined in the render state?

The way we have it now, we could end up with a ton of cached render states when the user resizes the canvas.